File: 프로그래머스/2/42584. 주식가격/주식가격.py
def solution(prices):
    answer = []
    n = len(prices)
    for i in range(n):
        cnt = 0
        for r in range(i, n - 1):
            if prices[i] <= prices[r+1]:
                cnt += 1
            else:
                cnt += 1
                break
        answer.append(cnt)

    return answer

File: 프로그래머스/2/42584. 주식가격/주식가격.java
import java.util.*;

class Solution {
    public int[] solution(int[] prices) {
        Deque<Integer> dq = new ArrayDeque<>();
        int N = prices.length;
        int[] arr = new int[N];
        for(int i = 0; i < N; i++){
            while(!dq.isEmpty() && prices[dq.peek()] > prices[i]){
                int idx = dq.pop();
                arr[idx] = i - idx;
            }
            dq.push(i);
        }
        while(!dq.isEmpty()){
            int idx = dq.pop();
            arr[idx] = N - idx - 1;
        }
        
        return arr;
    }
}